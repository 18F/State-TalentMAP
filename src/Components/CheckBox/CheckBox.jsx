import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { EMPTY_FUNCTION } from '../../Constants/PropTypes';

class CheckBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: { value: this.props.value },
    };
  }

  onCheck() {
    const { checked } = this.state;
    checked.value = !checked.value;
    this.setState({ checked },
      this.props.onCheckBoxClick(checked.value, { ...this.props }),
    );
  }

  render() {
    const { id, label, title, name } = this.props;
    const { checked } = this.state;
    return (
      <div className="usa-grid-full tm-checkbox">
        <input
          type="checkbox"
          id={id}
          title={title}
          name={name}
          value={checked.value}
          onChange={() => this.onCheck()}
          checked={checked.value}
        />
        <label htmlFor={id}>{label}</label>
      </div>
    );
  }
}

CheckBox.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
  onCheckBoxClick: PropTypes.func,
};

CheckBox.defaultProps = {
  onCheckBoxClick: EMPTY_FUNCTION,
};

export default CheckBox;
