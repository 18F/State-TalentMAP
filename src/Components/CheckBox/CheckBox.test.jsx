import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import CheckBox from './CheckBox';

describe('CheckBoxComponent', () => {
  it('can receive props', () => {
    const wrapper = shallow(
      <CheckBox
        id="id"
        label="label"
        title="title"
        name="name"
        value="value"
        onCheckBoxClick={() => {}}
      />,
    );
    expect(wrapper.instance().props.id).toBe('id');
  });

  it('can call the onCheck function upon selection', () => {
    const wrapper = shallow(
      <CheckBox
        id="id"
        label="label"
        title="title"
        name="name"
        value="value"
        onCheckBoxClick={() => {}}
      />,
    );
    wrapper.instance().onCheck();
    expect(wrapper.instance().state.checked.value).toBe(false);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <CheckBox
        id="id"
        label="label"
        title="title"
        name="name"
        value="value"
        onCheckBoxClick={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
