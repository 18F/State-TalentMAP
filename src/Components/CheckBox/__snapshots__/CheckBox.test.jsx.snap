// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxComponent matches snapshot 1`] = `
<div
  className="usa-grid-full tm-checkbox "
>
  <input
    checked="value"
    id="id"
    name="name"
    onChange={[Function]}
    title="title"
    type="checkbox"
    value="value"
  />
  <label
    htmlFor="id"
  >
    <span
      className=""
    >
      label
    </span>
  </label>
</div>
`;

exports[`CheckBoxComponent matches snapshot when small = true 1`] = `
<div
  className="usa-grid-full tm-checkbox tm-checkbox-small"
>
  <input
    checked="value"
    id="id"
    name="name"
    onChange={[Function]}
    title="title"
    type="checkbox"
    value="value"
  />
  <label
    htmlFor="id"
  >
    <span
      className=""
    >
      label
    </span>
  </label>
</div>
`;
